### PR TITLE
Remove support for future script macros

### PR DESCRIPTION
**Pull request by ksolo**
libxml2 added support for future script macros based on the below
specification

https://www.w3.org/TR/html401/appendix/notes.html#h-B.7.1

This syntax was reserved, but has yet to be defined. It is still unclear
how these macros will be used and what risks will be created if/when
implemented.

This PR removes support until more information is provided.

### CODE REVIEW

**Comment by flavorjones** [comment]
Hey @ksolo! Thanks so much for submitting this, I agree we should remove this support in Nokogiri.

It looks like this PR failed CI because the test you added is being run even when Nokogiri is built against the "system" libxml2 (in our CI, this is Ubuntu). We should skip this test unless we're running with the vendored libxml2.

An example of how to do this is in https://github.com/sparklemotion/nokogiri/blob/master/test/html/test_attributes.rb

Basically, skip the test unless `Nokogiri::VersionInfo.instance.libxml2? && Nokogiri::VersionInfo.instance.libxml2_using_system?`

Does that make sense?

**Comment by ksolo** [comment]
@flavorjones That makes perfect sense. Thank you!

**Comment by flavorjones** [comment]
:+1:
